app/src/main/java/org/schabi/newpipe/player/resolver/AudioPlaybackResolver.java

```diff
@@ -31,19 +38,28 @@ public AudioPlaybackResolver(@NonNull final Context context,
     @Override
     @Nullable
     public MediaSource resolve(@NonNull final StreamInfo info) {
-        final MediaSource liveSource = maybeBuildLiveMediaSource(dataSource, info);
+        final MediaSource liveSource = PlaybackResolver.maybeBuildLiveMediaSource(dataSource, info);
         if (liveSource != null) {
             return liveSource;
         }
 
-        final int index = ListHelper.getDefaultAudioFormat(context, info.getAudioStreams());
+        final List<AudioStream> audioStreams = new ArrayList<>(info.getAudioStreams());
+        removeTorrentStreams(audioStreams);
+
+        final int index = ListHelper.getDefaultAudioFormat(context, audioStreams);
         if (index < 0 || index >= info.getAudioStreams().size()) {
             return null;
         }
 
         final AudioStream audio = info.getAudioStreams().get(index);
         final MediaItemTag tag = StreamInfoTag.of(info);
-        return buildMediaSource(dataSource, audio.getUrl(), PlayerHelper.cacheKeyOf(info, audio),
-                MediaFormat.getSuffixById(audio.getFormatId()), tag);
+
+        try {
+            return PlaybackResolver.buildMediaSource(
+                    dataSource, audio, info, PlaybackResolver.cacheKeyOf(info, audio), tag);
+        } catch (final IOException e) {
+            Log.e(TAG, "Unable to create audio source:", e);
+            return null;
```

Comment:
It seems strange that this can return `null` but this **needs to be solved in a different PR**